src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1017,9 +1061,16 @@ class Simplex_tree {
    * \param simplex represent the simplex of which we search the star
    * \return Vector of Simplex_handle, empty vector if no cofaces found.
    */
-
-  Cofaces_simplex_range star_simplex_range(const Simplex_handle simplex) {
-    return cofaces_simplex_range(simplex, 0);
+  Star_simplex_range star_simplex_range(const Simplex_handle simplex) {
+    if constexpr (Options::link_nodes_by_label) {  // faster cofaces computation
+      Simplex_vertex_range rg = simplex_vertex_range(simplex);
+      std::vector<Vertex_handle> simp(rg.begin(), rg.end());
+      // must be sorted in decreasing order
+      assert(std::is_sorted(simp.begin(), simp.end(), std::greater<Vertex_handle>()));
+      return Star_simplex_range(Star_simplex_iterator(this, simp), Star_simplex_iterator());
+    } else {
+      return cofaces_simplex_range(simplex, 0);
+    }
   }
 
```

Comment:
`Cofaces_simplex_range cofaces_simplex_range(const Simplex_handle simplex, int codimension)` could also benefit from the optimization if we are able to track the heights of nodes we are crossing. In the meantime, we can have (eventhough it is not really pretty I admit...):
```C++
Cofaces_simplex_range cofaces_simplex_range(const Simplex_handle simplex, int codimension) {
    if constexpr (Options::link_nodes_by_label) {
       if (codimension == 0) return star_simplex_range(simplex);
       [usual code...]
    } else {
       [usual code...]
    }
}
```

Reply:
Ok, but this implies to cast `star_simplex_range` as a `cofaces_simplex_range`, which is taking some time (0.036 sec. vs 0.017 sec. without casting):
```cpp
      Star_simplex_range star_range(Star_simplex_iterator(this, std::move(simp)), Star_simplex_iterator());
      return Cofaces_simplex_range(star_range.begin(), star_range.end());
```

Reply:
Which benchmark? It is true that putting the handles in a vector is not free, but for a sparse complex, with the new option to use linked lists, it should save us from the expensive search on a large part of the complex.

By the way, you could skip `star_range` and pass the iterators directly to the vector (Cofaces_simplex_range).

Reply:
Benchmark is the one provided in this PR: https://github.com/VincentRouvreau/gudhi-devel/blob/fast_cofaces/src/Simplex_tree/benchmark/simplex_tree_stars_benchmark.cpp

I tried to skip `star_range`, but it is still slower (0.033 sec. vs 0.019 sec.). As you said, putting the handles in a vector is not free, but the return type is much understandable. I am ok to go with this solution as it will simplify the interfaces and the documentation. And you ?

Reply:
A vector is a simpler object, but if using the self-made iterators is faster (as it avoids filling a vector), why not make the cofaces an iterator too? I find it still weird that we have to different types "star_simplex_range" and "cofaces_simplex_range", when mathematically speaking there are the same thing. We just use one as special case, but I wouldn't be surprised if it confuses more than one user later. Can we not factorize something into the iterator? For example having two iterators, one for the codim==0, the other for codim>0, inheriting both from the same base class and they act differently depending on the value of Options::link_nodes_by_label. At the end, we would have a single type "cofaces_simplex_range" for both methods. It will be more work, but I think it will be cleaner.

Reply:
By the way, Clément made de star access an iterator, because I need to be able to iterate two stars at the same time and stop when the smallest star finishes (ie a complexity depending on the smallest star, not on the biggest) for the tower algorithm. Other applications could need something similar too, but for the star of a higher dimensional simplex. This would be in favor of the iterator range instead of the vector.

Reply:
star_simplex_range and coface_simplex_range now return the same type on [5104751](https://github.com/GUDHI/gudhi-devel/pull/877/commits/5104751ff686175d30b0089ddb20f6f21a108fa9)